src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/ExceptionRecordTest.java

```diff
@@ -38,4 +41,33 @@ public void should_upload_blob_and_create_exception_record() throws Exception {
             .pollInterval(500, TimeUnit.MILLISECONDS)
             .until(() -> Objects.equals(ProcessorEnvelopeStatusChecker.checkStatus(zipFileName), "COMPLETED"));
     }
+
+    @Test
+    public void should_dispatch_blob_and_create_exception_record_for_supplementary_evidence_with_ocr_classification()
+        throws Exception {
+        String zipFileName = ZipFileHelper.randomFileName();
+
+        var zipArchive = ZipFileHelper.createZipArchive(
+            singletonList("1111002.pdf"),
+            "supplementary_evidence_with_ocr_metadata.json",
+            zipFileName
+        );
+
+        await("File " + zipFileName + " should be dispatched")
+            .atMost(60, TimeUnit.SECONDS)
+            .pollInterval(500, TimeUnit.MILLISECONDS)
+            .until(() -> Objects.equals(RouterEnvelopesStatusChecker.checkStatus(zipFileName), "DISPATCHED"));
+
+        ProcessorEnvelopeResult[] processorEnvelopeResult = new ProcessorEnvelopeResult[1];
+        await("Exception record is created for " + zipFileName)
+            .atMost(60, TimeUnit.SECONDS)
+            .pollInterval(500, TimeUnit.MILLISECONDS)
+            .until(() -> (processorEnvelopeResult[0] = getCompletedEnvelope(zipFileName)) != null);
```

Comment:
I guess you don't need an array if you're using only one element

Reply:
Variable used in lambda expression should be final or effectively final.

Reply:
what a hack

Reply:
You can use some final object with one field and set that field. Or just make one more call after the waiting.

Reply:
which is what I am doing?

Reply:
But the fact that it's an array is a bit confusing to the person who reads the code, as the reason isn't immediately obvious.

Reply:
we will need different objects to assert and with this strategy we need another class just to hack this.it may be too much. we can move these awatings to new class which will return object and comment about array.?

Reply:
just make another call after await. we do this in most funcitonal tests.